swig/StringArray_API_extensions.i

```diff
@@ -0,0 +1,281 @@
+/**
+ * This SWIG interface extension provides support to
+ * allocate, return and manage arrays of strings through
+ * the class StringArray.
+ *
+ * This is then used to generate wrappers that return newly-allocated
+ * arrays of strings, so the user can them access them easily as a String[]
+ * on the Java side by a single call to StringArray::data(), and even manipulate
+ * them.
+ *
+ * It also implements working wrappers to:
+ *  - LGBM_BoosterGetEvalNames    (re-implemented with new API)
+ *  - LGBM_BoosterGetFeatureNames (original non-wrapped version didn't work).
+ * where the wrappers names end with "SWIG".
+ */
+
+// Use SWIG's `various.i` to get a String[] directly in one call:
+%apply char **STRING_ARRAY {char **StringArrayHandle_get_strings};
+
+#include <iostream>
+#include <memory>
+
+
+
+%inline %{
+
+    #define API_OK_OR_VALUE(api_return, return_value) if (api_return == -1) return return_value
+    #define API_OK_OR_NULL(api_return) API_OK_OR_VALUE(api_return, nullptr)
+
+    typedef void* StringArrayHandle;
+
+
+
+
+// TODO: @reviewer - I'd like to see the StringArray class moved to another .i file or to the src + include folders.
```

Comment:
either is possible, I think you can declare it in a .i file or keep it as hpp and reference it in the lightgbmlib.i file